### PR TITLE
[remoteopenhab] Use AbstractStorageBasedTypeProvider

### PR DESCRIPTION
**Pull request by lolodomo**
Related to #14954

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
Now tested and works well as expected using OH snapshot 3458.
Ready for review.

**Comment by lolodomo** [comment]
I changed channel types ids, that is not a good idea, I will revert to the original ids.

**Comment by lolodomo** [comment]
> I changed channel types ids, that is not a good idea, I will revert to the original ids.

Fixed.
Now ready again for a review.

**Comment by lolodomo** [comment]
@J-N-K : can you please have a look ?

**Comment by J-N-K** [comment]
I don't know much about the binding itself. Does it create a channel-type for each item or for each item-type?

**Comment by lolodomo** [comment]
> I don't know much about the binding itself. Does it create a channel-type for each item or for each item-type?

For each remote item type having specific state description. That means you can have several for item type Number in case one remote item was read-only and another one was not.
All this is unchanged by this PR. The only change is to use the new storage.

File: bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/RemoteopenhabChannelTypeProvider.java [hunk 1]
```diff
@@ -14,46 +14,69 @@
...
 @NonNullByDefault
-public class RemoteopenhabChannelTypeProvider implements ChannelTypeProvider {
-    private final List<ChannelType> channelTypes = new CopyOnWriteArrayList<>();
+public class RemoteopenhabChannelTypeProvider extends AbstractStorageBasedTypeProvider {
+
+    private static final String PATTERN_CHANNEL_TYPE_ID = "item%s%d";
+
+    private final Logger logger = LoggerFactory.getLogger(RemoteopenhabChannelTypeProvider.class);
+
+    private final Pattern channelTypeIdPattern = Pattern.compile("^item([a-zA-Z]+)([0-9]+)$");
```
Comment (J-N-K) [inline bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/RemoteopenhabChannelTypeProvider.java:54]: can be a `static` constant

**Review by J-N-K — CHANGES_REQUESTED** [review]
Otherwise looks good. I'm not sure some of the code ist most efficient, but I guess since new channel-types are only created on rare occasions (after the initial startup), it's fine.

File: bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/RemoteopenhabChannelTypeProvider.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/RemoteopenhabChannelTypeProvider.java]: You're right.
Changed.

**Review by J-N-K — APPROVED** [review]

**Pull request merged**